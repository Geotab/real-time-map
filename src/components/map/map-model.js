import L from "leaflet";
import { userAPI } from "../../services/api/user-api";
import { userInfo } from "../../dataStore/api-config";
import storage from "../../dataStore";
import { resetTransitionAnimation } from "../../utils/helper";

let MAPBOX;
try {
	MAPBOX = require("../../../map-box-token.key.json");
}
catch (ex) {
	MAPBOX = {
		ACCESS_TOKEN: false
	};
}

export const mapModel = {

	handleMapCreated(element) {
		storage.map = new L.Map(element, {
			doubleClickZoom: false,
			markerZoomAnimation: false,
			worldCopyJump: true
		});
		if (MAPBOX && MAPBOX.ACCESS_TOKEN) {
			mapModel.addMapBoxTileLayer(storage.map);
		}
		else {
			mapModel.addOSMTileLayer(storage.map);
		}

		storage.map.on("mousedown", () => {
			storage.map.closePopup();
		});
		storage.map.on("zoom", resetTransitionAnimation);
	},

	addMapBoxTileLayer: map => {
		L.tileLayer(`https://api.tiles.mapbox.com/v4/{id}/{z}/{x}/{y}.png?access_token=${MAPBOX.ACCESS_TOKEN}`, {
			attribution: "Map data &copy; <a href=\"https://www.openstreetmap.org/\">OpenStreetMap</a> contributors, <a href=\"https://creativecommons.org/licenses/by-sa/2.0/\">CC-BY-SA</a>, Imagery Â© <a href=\"https: //www.mapbox.com/\">Mapbox</a>",
			minZoom: storage.minZoom,
			maxZoom: storage.maxZoom,
			id: "mapbox.streets",
			accessToken: MAPBOX.ACCESS_TOKEN
		}).addTo(map);
	},

	addOSMTileLayer(map) {
		L.tileLayer("http://{s}.tile.openstreetmap.org/{z}/{x}/{y}.png", {
			attribution: '&copy; <a href="https://www.openstreetmap.org/copyright">OpenStreetMap</a>',
			subdomains: ["a", "b", "c"],
			minZoom: storage.minZoom,
			maxZoom: storage.maxZoom,
		}).addTo(map);
	},

	setMapToCompanyAddress() {
		return userAPI.getUserInfo(userInfo.userName)
			.then(result => {
				userAPI.getCoordinatesFromAddress(result[0].companyAddress)
					.then(result =>
						mapModel.mapSetView(L.latLng(result[0].y, result[0].x))
					)
					.catch(() => mapModel.mapSetView(0));
			})
			.catch(() => alert("Server is unavailable, please try again later."));
	},

	mapSetView(latlng) {
		if (latlng.length === 0) {
			storage.map.setView([43.515228, -79.683523], 12);
			console.log("Default location set to user location.");
		}
		else {
			storage.map.setView(latlng, 12);
			console.log("Location set to company address.");
		}
	}

};
