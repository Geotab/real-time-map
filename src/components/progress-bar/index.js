export { progressBar } from "./progress-indicator";