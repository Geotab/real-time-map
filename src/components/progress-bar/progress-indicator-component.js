import React, { Component } from "react";

export const ProgressIndicatorComponent = () => (
  <div id="progressbar">
    <div role="progressbar" className="mdc-linear-progress ">
      <div className="mdc-linear-progress__buffering-dots"></div>
      <div className="mdc-linear-progress__buffer"></div>
      <div className="mdc-linear-progress__bar mdc-linear-progress__primary-bar">
        <span className="mdc-linear-progress__bar-inner"></span>
      </div>
      <div className="mdc-linear-progress__bar mdc-linear-progress__secondary-bar">
        <span className="mdc-linear-progress__bar-inner"></span>
      </div>
    </div>
  </div>
);
