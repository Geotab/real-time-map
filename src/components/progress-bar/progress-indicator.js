import { MDCLinearProgress } from "@material/linear-progress";

export const progressBar = {
  update(progress) {
    const linearProgress = new MDCLinearProgress(document.querySelector(".mdc-linear-progress"));
    linearProgress.open();
    linearProgress.progress = progress;
    if (progress >= 1) {
      setTimeout(() => linearProgress.close(), 1000);
    }
  }
};