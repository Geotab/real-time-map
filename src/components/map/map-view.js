import React, { Component } from "react";
import { mapModel } from "./map-model";
import { progressBar } from "../progress-bar";
import { ProgressIndicatorComponent } from "../progress-bar/progress-indicator-component";
import { SnackBar } from "../snackbar/snackbar-component";

export class MapView extends React.Component {
  componentDidMount() {
    mapModel.handleMapCreated("RTM-Map");
    progressBar.update();
  }

  render() {
    return (
      <div id="RTM-Map-Container">
        <div id="RTM-Map">
          <ProgressIndicatorComponent />
          <button
            type="button"
            className="collapsible"
            id="collapse-button"
            title="Open Configuration Panel"
          ></button>
          <SnackBar />
        </div>
      </div>
    );
  }
}
