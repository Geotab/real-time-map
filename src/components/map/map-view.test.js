import React from "react";
import {MapView} from "./map-view";
import { render, mount, shallow } from "enzyme";
import {mapModel} from "./map-model";
import { progressBar } from "../progress-bar";

import {
  spyOnFunction,
  spyOnAccessorFunction,
  mockMap,
  mockDateKeeper,
  createDivsWithID,
  createDivsWithClasses
} from "../../../tests/utils/test-helpers";

let wrapper;

test("should successfully render the map component", () => {
  const component = render(<MapView />);
  expect(component).toMatchSnapshot();
});

test("should successfully call the component did mount when rendering view", () => {
  spyOnFunction(progressBar, "update");
  mapModel.handleMapCreated = jest.fn();
  wrapper = shallow(<MapView />);
  expect(mapModel.handleMapCreated).toHaveBeenCalledWith("RTM-Map");
});