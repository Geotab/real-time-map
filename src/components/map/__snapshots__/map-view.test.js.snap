// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should successfully render the map component 1`] = `
<div
  id="RTM-Map-Container"
>
  <div
    id="RTM-Map"
  >
    <div
      id="progressbar"
    >
      <div
        class="mdc-linear-progress "
        role="progressbar"
      >
        <div
          class="mdc-linear-progress__buffering-dots"
        />
        <div
          class="mdc-linear-progress__buffer"
        />
        <div
          class="mdc-linear-progress__bar mdc-linear-progress__primary-bar"
        >
          <span
            class="mdc-linear-progress__bar-inner"
          />
        </div>
        <div
          class="mdc-linear-progress__bar mdc-linear-progress__secondary-bar"
        >
          <span
            class="mdc-linear-progress__bar-inner"
          />
        </div>
      </div>
    </div>
    <button
      class="collapsible"
      id="collapse-button"
      title="Open Configuration Panel"
      type="button"
    />
    <div
      class="mdc-snackbar mdc-snackbar--align-start"
    >
      <div
        class="mdc-snackbar__surface"
      >
        <div
          aria-live="polite"
          class="mdc-snackbar__label"
          role="status"
        />
        <div
          class="mdc-snackbar__actions"
        >
          <button
            class="mdc-button mdc-snackbar__action"
            type="button"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
